answers.md

```diff
@@ -0,0 +1,50 @@
+# Answers
+## HPC good citizenship
+1. No. `-pe openmp 64` only tells the grid engine to allocate 64 cores. But the real number of processors used are determined by the programs running on it.
+
+
+2. Review the scripts before submitting to the grid engine. Softwares written by others usually have documentations. If not, try running the softwares with a smaller dataset as a test.
+
+
+3. Use following command
+
+```
+/usr/bin/time -f "maximum resident set size = %M, elapsed real time = %E, cpu time = %S, user time = %U" command
+```
+The idea is from this [page](https://unix.stackexchange.com/questions/375889/unix-command-to-tell-how-much-ram-was-used-during-program-runtime).
+
+
+4. For the maximum resident set size, Kbytes. For the elapsed real time, [hours:]minutes:seconds. For cpu time and user time, seconds.
+
+
+5. Use following commands
+```
+# Checking that a file exists
+[ -e filename ]
+# Checking that a file exists and is not empty
+[ -s filename ]
+```
+
+
+6. Use following command
+```
+[ -e filename ] || run-a-job
```

Comment:
Hi here you should just do if [ ! -e test.txt ] as this checks if something is not there. typically in CS "!" means not. you could also just do [ -e test.txt ] || CMD and make it a one liner